helm/ocs-monkey-generator/templates/deployment.yaml

```diff
@@ -64,3 +64,59 @@ spec:
         - name: logdir
           persistentVolumeClaim:
             claimName: '{{ template "ocs-monkey-generator.fullname" . }}'
+
+---
+{{- if eq .Values.runChaosRunner true }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: '{{ .Values.chaosName }}'
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: '{{ include "ocs-monkey-generator.name" . }}'
+      app.kubernetes.io/instance: "{{ .Release.Name }}"
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: '{{ include "ocs-monkey-generator.name" . }}'
+        app.kubernetes.io/instance: "{{ .Release.Name }}"
```

Comment:
I don't think you can use the same set of matchLabels in both deployments or the Deployment and/or ReplicaSet controller will likely get confused.
Suggestion: You could use `'{{ include "ocs-monkey-generator.name" . }}-chaos'`